File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int maxPro = 0;
        int minPrice = prices[0];
        for(int i = 1; i<n; i++){
            int cost = prices[i] - minPrice;
            minPrice = min(prices[i],minPrice);
            maxPro = max(maxPro,cost);
        }
        return maxPro;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int minBuy = prices[0];
        int n = prices.length;
        for(int i = 1; i<n; i++){
            profit = Math.max(profit, prices[i] - minBuy);
            minBuy = Math.min(minBuy,prices[i]);
        }
        return profit;
    }
}